src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,297 @@
+            isResponsesVisible.put(participantType.name(), true);
+        }
+        visibilitySettings.setResponseVisibleFor(isResponsesVisible);
+        
+        feedbackPathSettings.setNumberOfEntitiesToGiveFeedbackToChecked(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS);
+        feedbackPathSettings.setNumOfEntitiesToGiveFeedbackToValue(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS ?
+                                                                     1 :
+                                                                     question.numberOfEntitiesToGiveFeedbackTo);
+        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
+        
+        visibilitySettings.setVisibilityMessages(question.getVisibilityMessage());
+        
+        qnForm.setQuestionSpecificEditFormHtml(question.getQuestionDetails().getQuestionSpecificEditFormHtml(question.questionNumber));
+        qnForm.setEditable(false);
+        
+        qnForms.add(qnForm);
+    }
+
+
+    private void buildNewQuestionForm(FeedbackSessionAttributes feedbackSession,
+                                      List<FeedbackQuestionAttributes> questions) {
+        newQnForm = new FeedbackQuestionEditForm();
+        newQnForm.setDoneEditingLink(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                                        .withUserId(account.googleId)
+                                        .withCourseId(feedbackSession.courseId)
+                                        .withSessionName(feedbackSession.feedbackSessionName));
+        
+        newQnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_ADD));
+        newQnForm.setCourseId(feedbackSession.courseId);
+        newQnForm.setFeedbackSessionName(feedbackSession.feedbackSessionName);
+        newQnForm.setQuestionNumberSuffix("");
+        
+        newQnForm.setQuestionTypeOptions(getQuestionTypeChoiceOptions());
+      
+        FeedbackQuestionFeedbackPathSettings feedbackPathSettings = new FeedbackQuestionFeedbackPathSettings();
+        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings();
+        
+        newQnForm.setFeedbackPathSettings(feedbackPathSettings);
+        newQnForm.setVisibilitySettings(visibilitySettings);
+        
+        feedbackPathSettings.setGiverParticipantOptions(getParticipantOptions(null, true));
+        feedbackPathSettings.setRecipientParticipantOptions(getParticipantOptions(null, false));
+        feedbackPathSettings.setNumOfEntitiesToGiveFeedbackToValue(1);
+        
+        newQnForm.setNumOfQuestionsOnPage(questions.size());
+        newQnForm.setQuestionNumberOptions(getQuestionNumberOptions(questions.size() + 1));
+      
+        newQnForm.setQuestionSpecificEditFormHtml(getNewQuestionSpecificEditFormHtml());
+        newQnForm.setEditable(true);
```

Comment:
Would it be a good idea to extract this into `FeedbackQuestionEditForm` as some default form of it?